state-synchronizer/src/synchronizer.rs

```diff
@@ -104,6 +107,22 @@ impl StateSynchronizer {
             .await?;
         cb_receiver.await?
     }
+
+    pub async fn add_reconfig_subscribers(
+        &self,
+        reconfig_event_subscriptions: Vec<Box<dyn ReconfigurationEventSubscription>>,
+    ) -> Result<()> {
+        let mut sender = self.coordinator_sender.clone();
+        if let Err(e) = sender
+            .send(CoordinatorMessage::ReconfigurationEventSubscriptions(
+                reconfig_event_subscriptions,
+            ))
+            .await
+        {
+            return Err(format_err!(e));
+        }
+        Ok(())
+    }
```

Comment:
do we foresee any big reasons to change the reconfig subscribers at runtime? why not just pass in this list of subscribers at construction? that would simplify the startup flow I think.